vtkLookingGlassInterface.cxx

```diff
@@ -627,6 +628,132 @@ void vtkLookingGlassInterface::GetTilePosition(int tile, int pos[2])
   pos[1] = (tile / this->QuiltTiles[0]) * this->RenderSize[1];
 }
 
+void vtkLookingGlassInterface::RenderQuilt(
+  vtkOpenGLRenderWindow* rw, std::function<void(void)>* renderFunc)
+{
+  vtkCollectionSimpleIterator rsit;
+
+  // loop over the tiles, render,and blit
+  vtkOpenGLFramebufferObject* renderFramebuffer;
+  vtkOpenGLFramebufferObject* quiltFramebuffer;
+  this->GetFramebuffers(rw, renderFramebuffer, quiltFramebuffer);
+
+  auto ostate = rw->GetState();
+
+  ostate->PushFramebufferBindings();
+  renderFramebuffer->Bind(GL_READ_FRAMEBUFFER);
+
+  int renderSize[2];
+  this->GetRenderSize(renderSize);
+
+  int tcount = this->GetNumberOfTiles();
+
+  // save the original camera settings
+  auto* renCol = rw->GetRenderers();
+  vtkRenderer* aren;
+  std::vector<vtkCamera*> Cameras;
+  for (renCol->InitTraversal(rsit); aren = renCol->GetNextRenderer(rsit);)
+  {
+    // Ugly piece of code - we need to know if the camera already
+    // exists or not. If it does not yet exist, we must reset the
+    // camera here - otherwise it will never be done (missing its
+    // oppportunity to be reset in the Render method of the
+    // vtkRenderer because it will already exist by that point...)
+    if (!aren->IsActiveCameraCreated())
+    {
+      aren->ResetCamera();
+    }
+    auto oldCam = aren->GetActiveCamera();
+    oldCam->SetLeftEye(1);
+    oldCam->Register(rw);
+    Cameras.push_back(oldCam);
+    vtkNew<vtkCamera> newCam;
+    aren->SetActiveCamera(newCam);
+  }
+
+  // Make sure we can use offscreen buffers
+  auto prevUseOffScreenBuffers = rw->GetUseOffScreenBuffers();
+  rw->UseOffScreenBuffersOn();
+
+  // loop over all the tiles and render then and blit them to the quilt
+  for (int tile = 0; tile < tcount; ++tile)
+  {
+    renderFramebuffer->Bind(GL_DRAW_FRAMEBUFFER);
+    ostate->vtkglViewport(0, 0, renderSize[0], renderSize[1]);
+    ostate->vtkglScissor(0, 0, renderSize[0], renderSize[1]);
+
+    int count = 0;
+    for (renCol->InitTraversal(rsit); aren = renCol->GetNextRenderer(rsit); ++count)
+    {
```

Comment:
Looping over all renderers wasn't done in `vtkLookingGlassPass`, but I assumed it is safe to do there too. Let me know if that is not the case, and we can figure out something different.

Reply:
Yes a pass is set on a renderer, not renderwindow. I'm not sure we support multiple viewports on looking glass so it may be moot.

Reply:
One idea is that I could add another optional argument to `RenderQuilt()` for `renCol` (the `vtkRendererCollection*` to use, that defaults to `nullptr`, and if it is `nullptr`, then grab it via `GetRenderers()` from the render window). The `vtkLookingGlassPass` can set this to be just the single renderer.

Or if you think it's moot, then I can also leave things the way they are :slightly_smiling_face:

Reply:
This has been fixed. The looking glass pass now only uses its one renderer.